src/KafkaW/ProducerTopic.cpp

```diff
@@ -17,8 +17,7 @@ ProducerTopic::ProducerTopic(std::shared_ptr<Producer> ProducerPtr,
     : KafkaProducer(ProducerPtr), Name(std::move(TopicName)) {
 
   std::string ErrStr;
-  RdKafkaTopic = std::unique_ptr<RdKafka::Topic>(RdKafka::Topic::create(
-      KafkaProducer->getRdKafkaPtr(), Name, ConfigPtr.get(), ErrStr));
+  RdKafkaTopic = KafkaProducer->createTopic(Name, ErrStr);
```

Comment:
use this `RdKafkaTopic` in the call to `produce()`, calling `createTopic` every time in `Prodcuer::produce` should be avoided to maintain performance.